tests/api2/test_crud_events.py

```diff
@@ -0,0 +1,135 @@
+import contextlib
+import threading
+import typing
+
+from middlewared.test.integration.assets.crypto import get_cert_params, root_certificate_authority
+from middlewared.test.integration.utils import call
+from middlewared.test.integration.utils.client import client
+
+
+def event_thread(event_endpoint: str, context: dict):
+    with client(py_exceptions=False) as c:
+        context['start_event'].set()
+        subscribe_payload = c.event_payload()
+        event = subscribe_payload['event']
+        context['event'] = event
+
+        def cb(mtype, **message):
+            if len(message) != 3 or not all(
```

Comment:
Will that be a problem if another extra key is present in the message?

Reply:
That won't happen for CRUD automatic events ?
And if it does, this needs to be updated as well to account for that

Reply:
It is to avoid other system events which would be emitting from middleware. There are certain APIs which send the same event but under different circumstances so like `replication.query` or `disk.query` send the same query event but it is not sent by middleware on automatic CRUD operations.